lib/loader/index.js

```
@@ -0,0 +1,254 @@
+  }
+
+  if (callback && typeof callback !== 'function') {
+    throw new Error('`callback` must be a function');
+  }
+}
+
+module.exports = {
+  /**
+   * Load a script from a URL
+   *
+   * The script is loaded on a later turn of the event loop. If a callback is
+   * provided, it will be invoked on success in all supported browsers. The
+   * callback will execute after the script.
+   *
+   * In modern IE and other browsers, the callback will be invoked with an error
+   * on failure as well. A timeout in milliseconds may be provided as
+   * `options.timeout`. If this timeout expires before loading fails or
+   * succeeds, the callback will be invoked with an error.
+   *
+   * @param {string} url - URL of the script
+   * @param {object} [options]
+   * @param {object} options.attributes - an object containing key/value pairs
+   *  to be used as attributes on the created `script` element (optional)
+   * @param {number} options.timeout - timeout in milliseconds (defaults to
+   *  10000ms)
+   * @param {function} [callback] - node-style callback
+   */
+  loadScript: function (url, options, callback) {
+    if (typeof options === 'function') {
+      callback = options;
+      options = null;
+    }
+
+    options = options || {};
+    options.timeout = options.timeout || DEFAULT_TIMEOUT;
+
+    validateArguments(url, options, callback);
+
+    var script = doc.createElement('script');
+    var done = false;
+    var attr;
+
+    if (options.attributes) {
+      for (attr in options.attributes) {
+        script.setAttribute(attr, options.attributes[attr]);
+      }
+    }
+
+    function cleanUp (err) {
+      done = true;
+      clearTimeout(timeoutHandle);
+      script.onload = script.onreadystatechange = script.onerror = null;
+      script.parentNode.removeChild(script);
+
+      if (callback) {
+        try {
+          callback(err);
+        }
+        catch (e) {
+          // FIXME: report this error once analytics support is available
```

Maybe not applicable to this PR specifically, but surely there's something we can do to avoid swallowing these exceptions completely? I'm happy to open a PR if someone has an idea about how to do that (logging, wrapping the error, etc.).